packages/ui/src/calendar/date-calendar/index.stories.tsx

```diff
@@ -0,0 +1,525 @@
+					} }
+					disabled={ [
+						{
+							// Disable any date before today
+							before: new TZDate( new Date(), args.timeZone ),
+						},
+					] }
+				/>
+
+				<p>
+					Calendar set to { args.timeZone ?? 'current' } timezone, disabling selection for all dates
+					before today, and starting with a default date of 1 week from today.
+				</p>
+			</>
+		);
+	},
+	args: {
+		timeZone: 'Pacific/Auckland',
+	},
+	argTypes: {
+		selected: {
+			control: false,
+		},
+		defaultSelected: {
+			control: false,
+		},
+		disabled: {
+			control: false,
+		},
+	},
+};
+
+const today = new Date();
+const oneWeekBefore = ( date: Date ) => {
+	const toReturn = new Date( date );
+	toReturn.setDate( date.getDate() - 7 );
+	return toReturn;
+};
+const startOfMonth = ( date: Date ) => new Date( date.getFullYear(), date.getMonth(), 1 );
+const oneMonthBefore = ( date: Date ) => {
+	const toReturn = new Date( date );
+	toReturn.setMonth( date.getMonth() - 1 );
+	return toReturn;
+};
+const startOfYear = ( date: Date ) => new Date( date.getFullYear(), 0, 1 );
+const oneYearBefore = ( date: Date ) => {
+	const toReturn = new Date( date );
+	toReturn.setFullYear( date.getFullYear() - 1 );
+	return toReturn;
+};
+
+function computePresetRange( preset: string ) {
+	let targetDate;
+	switch ( preset ) {
+		case 'today':
+			targetDate = today;
+			break;
+		case 'one-week-ago':
+			targetDate = oneWeekBefore( today );
+			break;
+		case 'start-of-month':
+			targetDate = startOfMonth( today );
+			break;
+		case 'one-month-ago':
+			targetDate = oneMonthBefore( today );
+			break;
+		case 'start-of-year':
+			targetDate = startOfYear( today );
+			break;
+		case 'one-year-ago':
+			targetDate = oneYearBefore( today );
+			break;
+	}
+	return targetDate;
+}
+
+function PresetItem( {
+	id,
+	label,
+	selected,
+	setSelected,
+}: {
+	id: string;
+	label: string;
+	selected: boolean;
+	setSelected: ( id: string ) => void;
+} ) {
+	return (
+		<Composite.Item
+			render={ <Button __next40pxDefaultSize variant={ selected ? 'primary' : undefined } /> }
+			id={ id }
+			onClick={ () => setSelected( id ) }
+			role="option"
+			aria-selected={ selected ? 'true' : undefined }
+			className="preset-listbox__item"
+			// It's fine to autofocus when the popover opens
+		>
+			<span className="preset-listbox__item-label">
+				{ label }
+				<span
+					className="preset-listbox__item-label-bullet"
+					style={ { visibility: selected ? 'visible' : 'hidden' } }
+				>
+					&bull;
+				</span>
+			</span>
+		</Composite.Item>
+	);
+}
+
+/**
+ * The component can be used in controlled mode. This is useful, for example,
+ * when in need of keeping the component in sync with external input fields or
+ * more advanced UI patterns such as a preset selector.
+ *
+ * _Note: this example doesn't handle time zones, and is not responsive. Do not
+ * assume this is production-ready code._
+ */
+export const WithPresetsDialog: Story = {
```

Comment:
@ciampo What should we do with stories like this, that use a lot of `@wordpress/components`? It's kind of bad for isolation testing, since we need to load the `@wordpress/components` stylesheet for them to display correctly.

Would it be reasonable to move them to the reference site, if they're mainly meant as documentation of a pattern? In any case it seems like a lot of code to maintain for a story.

Reply:
I agree with everything.

I think this example is of high value in providing a real-world example of how to integrate the component. Moving it to the reference site makes sense.